qiita_pet/uimodules/prep_template_tab.py

```diff
@@ -0,0 +1,273 @@
+# -----------------------------------------------------------------------------
+# Copyright (c) 2014--, The Qiita Development Team.
+#
+# Distributed under the terms of the BSD 3-clause License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# -----------------------------------------------------------------------------
+
+from operator import itemgetter
+from os.path import basename
+from collections import defaultdict
+
+from future.utils import viewitems
+
+from qiita_db.util import (get_filetypes, get_files_from_uploads_folders,
+                           get_data_types, convert_to_id, get_filepath_types)
+from qiita_db.study import Study
+from qiita_db.data import RawData
+from qiita_db.ontology import Ontology
+from qiita_db.metadata_template import (PrepTemplate, TARGET_GENE_DATA_TYPES,
+                                        PREP_TEMPLATE_COLUMNS_TARGET_GENE)
+from qiita_db.parameters import (Preprocessed454Params,
+                                 PreprocessedIlluminaParams)
+from qiita_pet.util import STATUS_STYLER
+from qiita_pet.handlers.util import download_link_or_path
+from .base_uimodule import BaseUIModule
+
+
+filepath_types = [k.split('_', 1)[1].replace('_', ' ')
+                  for k in get_filepath_types()
+                  if k.startswith('raw_')]
+fp_type_by_ft = defaultdict(
+    lambda: filepath_types, SFF=['sff'], FASTA=['fasta', 'qual'],
+    FASTQ=['barcodes', 'forward seqs', 'reverse seqs'])
+
+
+def get_accessible_raw_data(user):
```

Comment:
I agree that we should move this away from here. However, I'm afraid of spending time in something that will be removed soon. There are some issues that make me think that:
 - One is https://github.com/biocore/qiita/issues/1145 which englobes changes like this.
 - We should provide finer control on the study elements ownership and visibility. I sent this idea in an email but we've not been discussing it. Related to https://github.com/biocore/qiita/issues/1215 and https://github.com/biocore/qiita/issues/1193.

I agree that we should improve all the code as we move forward, but I think it is not worth to spend time in moving this code around (it was already like this) if this is going to be refactored.